xarray/tests/test_dataarray.py

```diff
@@ -3000,9 +3000,92 @@ def test_to_and_from_iris(self):
         auto_time_dimension = DataArray.from_iris(actual)
         assert auto_time_dimension.dims == ('distance', 'dim_1')
 
-        actual.coord('distance').var_name = None
-        with raises_regex(ValueError, 'no var_name attribute'):
-            DataArray.from_iris(actual)
+        # use var_name
```

Comment:
please split this into a few smaller test methods -- that makes errors easier to track down.

Reply:
Agree. I will group iris-conversion-related tests in a separate class (following pytest conventions, and without inheriting from `TestCase`):

```python
class TestIrisConversion(object):
    def test_to_iris(self):
        ...

    def test_from_iris(self):
        ...

    def test_name_conversion(self):
        ...

```

Reply:
Sounds good! If you write that way, see if you can simplify things a little bit by using `pytest.mark.parametrize`.